### PR TITLE
Bluetooth: Mesh: Use IS_ENABLE replace #if

### PR DESCRIPTION
**Pull request by LingaoM**
1. Use IS_ENABLED replace #if defined to make code more readable
2. Move adv_tag_to_str to send_pending_adv, since only this used.

### CODE REVIEW

File: subsys/bluetooth/mesh/adv_ext.c [hunk 1]
```diff
@@ -71,33 +71,26 @@ static bool schedule_send(struct bt_mesh_ext_adv *ext_adv);
...
-#if defined(CONFIG_BT_MESH_ADV_EXT_RELAY_USING_MAIN_ADV_SET)
-			BT_MESH_ADV_TAG_BIT_RELAY |
-#endif /* CONFIG_BT_MESH_ADV_EXT_RELAY_USING_MAIN_ADV_SET */
-#if defined(CONFIG_BT_MESH_PB_ADV)
-			BT_MESH_ADV_TAG_BIT_PROV |
-#endif /* CONFIG_BT_MESH_PB_ADV */
-			BT_MESH_ADV_TAG_BIT_LOCAL
+		.tags = ( BT_MESH_ADV_TAG_BIT_LOCAL |
+			  IS_ENABLED(CONFIG_BT_MESH_ADV_EXT_FRIEND_SEPARATE) ?
+				0 : BT_MESH_ADV_TAG_BIT_FRIEND |
```
Comment (PavelVPV) [inline subsys/bluetooth/mesh/adv_ext.c:76]: Are you sure parentheses are not needed here? Ternary operator has lower precedence than bitwise OR: https://en.cppreference.com/w/c/language/operator_precedence

Comment (LingaoM) [reply subsys/bluetooth/mesh/adv_ext.c]: I'm not sure, maybe look preprocessing .i file, i still don't know how to view the .i file compiled by zephyr. But add parentheses always right.

File: subsys/bluetooth/mesh/adv_ext.c [hunk 2]
```diff
@@ -71,69 +71,49 @@ static bool schedule_send(struct bt_mesh_ext_adv *ext_adv);
...
 
 static inline struct bt_mesh_ext_adv *gatt_adv_get(void)
 {
-	if (IS_ENABLED(CONFIG_BT_MESH_ADV_EXT_GATT_SEPARATE)) {
-		return &advs[ARRAY_SIZE(advs) - 1];
-	} else {
-		return &advs[0];
-	}
+	return IS_ENABLED(CONFIG_BT_MESH_ADV_EXT_GATT_SEPARATE) ?
+			&advs[ARRAY_SIZE(advs) - 1] : &advs[0];
```
Comment (jhedberg) [inline subsys/bluetooth/mesh/adv_ext.c:116]: I know you're a fan of compact code, however I really don't think this improves readability. The C ternary operator is useful in places where you can't use `if ()` however I don't think it improves readability at all (IMO it does the contrary).

File: subsys/bluetooth/mesh/adv_ext.c [hunk 3]
```diff
@@ -71,69 +71,49 @@ static bool schedule_send(struct bt_mesh_ext_adv *ext_adv);
...
 	     "Insufficient adv instances");
 
 static inline struct bt_mesh_ext_adv *relay_adv_get(void)
 {
-	if (!!(CONFIG_BT_MESH_RELAY_ADV_SETS)) {
-		return &advs[1];
-	} else {
-		return &advs[0];
-	}
+	return !!(CONFIG_BT_MESH_RELAY_ADV_SETS) ? &advs[1] : &advs[0];
```
Comment (jhedberg) [inline subsys/bluetooth/mesh/adv_ext.c:110]: Same here, and this is also out of scope of the PR, since it's talking about replacing `#if` and not `if ()`

**Comment by omkar3141** [comment]
IMO the old code was more readable.
